circular-tour.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import "fmt"
+
+func main() {
+
+	petrol := []int{4, 6, 7, 4}
+	distance := []int{6, 5, 3, 5}
+
+	prev, curr := 0, 0
```

Comment:
what exactly `prev` is storing ??

Reply:
`prev` is storing amount of petrol below zero when travel from one point to another. Remaining petrol after completing the cycle + negative petrol during the whole cycle should be greater than equal to zero to consider a point as a starting point for ans.